source/extensions/common/wasm/wavm/wavm.cc

```diff
@@ -276,8 +276,16 @@ bool Wavm::initialize(const std::string& code, absl::string_view name, bool allo
 
 void Wavm::start(Context* context) {
   auto f = getStartFunction(moduleInstance_);
-  if (!f)
+  if (f) {
+    CALL_WITH_CONTEXT(invokeFunctionChecked(context_, f, {}), context);
+  }
+  if (emscriptenInstance_) {
+    Emscripten::initializeGlobals(context_, irModule_, moduleInstance_);
+  }
+  f = asFunctionNullable(getInstanceExport(moduleInstance_, "main"));
```

Comment:
Actually, on a second thought, what if `main()` is exported as the start function? Could we guard against that and not call it twice?

Reply:
AFAICT the "start" function is special and according to the docs should be called (if available) before any exported functions e.g. "main" or "_main".  In this case it is also called before the Emscripten module globals are initialized because that is what wavm-run does and this is WAVM specific code.

Reply:
Yeah, my point was that `main()` function might be marked as the "start" function (at the WASM level), in which case we'd call it twice... but maybe it wouldn't be exported then, though.

Anyway, we can deal with it if that becomes an issue with other languages.